internal/pkg/agent/application/coordinator/coordinator.go

```diff
@@ -683,10 +683,16 @@ func (c *Coordinator) Run(ctx context.Context) error {
 	// shutdown state.
 	defer close(c.stateBroadcaster.InputChan)
 
-	// The usual state refresh happens in the main run loop in Coordinator.runner,
-	// so before/after the runner call we need to trigger state change broadcasts
-	// manually with refreshState.
-	c.setCoordinatorState(agentclient.Starting, "Waiting for initial configuration and composable variables")
+	if c.varsMgr != nil {
+		// The usual state refresh happens in the main run loop in Coordinator.runner,
+		// so before/after the runner call we need to trigger state change broadcasts
+		// manually with refreshState.
```

Comment:
This comment is talking about the `c.refreshState()` call on line 696. This should really stay above that line.